confgen/api_internal.go

```diff
@@ -775,25 +782,46 @@ func populateVolumeGroup(confMap conf.ConfMap, globalVolumeGroupMap volumeGroupM
 			return
 		}
 
```

Comment:
It seems like this function should be reorganized to take advantage of IsVolumeGroupSharedSMB() and IsVolumeGroupSharedNFS()

Perhaps it would simplify the function

Reply:
I could certainly add `IsVolumeGroupSharedViaSMB` and `IsVolumeGroupSharedViaNFS` bools to the `VolumeGroup` struct and handle it that way.

That would _probably_ be an improvement.  It adds some "distributed state", where the flag gets set in one place and checked in another, compared to "get the list of SMB volume groups and pass those to `createSMBConf()`", which may be clearer.  I could go either way.

Truthfully, I was just trying to get `etcd-mgmt` working with the minimal amount of changes to `confgen`.  Sorting all the issues between `monitord` and `confgen` took quite awhile ...